sync/class.jetpack-sync-module-plugins.php

```diff
@@ -30,68 +31,129 @@ public function init_before_send() {
+		}
+
+		// for plugin installer
+		if ( empty( $plugins ) && method_exists( $upgrader, 'plugin_info' ) ) {
+			$plugins = array( $upgrader->plugin_info() );
+		}
 
-			if ( is_wp_error( $errors) ) {
-				foreach ( $details['plugins'] as $slug ) {
+		if ( empty( $plugins ) ) {
+			return; // We shouldn't be here
+		}
+
+		switch ( $details['action'] ) {
+			case 'update':
+				$state = array(
+					'is_autoupdate' => Jetpack_Constants::is_true( 'JETPACK_PLUGIN_AUTOUPDATE' ),
+				);
+				$errors = $this->get_errors( $upgrader->skin );
+				if ( $errors ) {
+					foreach ( $plugins as $slug ) {
+						/**
+						 * Sync that a plugin update failed
+						 *
+						 * @since 5.8.0
+						 *
+						 * @module sync
+						 *
+						 * @param string $plugin, Plugin slug
+						 * @param string Error code
+						 * @param string Error message
+						 */
+						do_action( 'jetpack_plugin_update_failed', $this->get_plugin_info( $slug ), $errors['code'], $errors['message'], $state );
+					}
+					return;
+				}
+				/**
+				 * Sync that a plugin update
+				 *
+				 * @since 5.8.0
+				 *
+				 * @module sync
+				 *
+				 * @param array() $plugin, Plugin Data
+				 */
+				do_action( 'jetpack_plugins_updated', array_map( array( $this, 'get_plugin_info' ), $plugins ), $state );
+				break;
+			case 'install':
+
+		}
+
+		if ( 'install' === $details['action'] ) {
+			$errors = $this->get_errors( $upgrader->skin );
+			if ( $errors ) {
+				foreach ( $plugins as $slug ) {
 					/**
 					 * Sync that a plugin update failed
 					 *
 					 * @since 5.8.0
-					 * 
+					 *
 					 * @module sync
 					 *
 					 * @param string $plugin, Plugin slug
 					 * @param string Error code
 					 * @param string Error message
 					 */
-					do_action( 'jetpack_plugin_update_failed', $slug, $errors->get_error_code(), $errors->get_error_message() );
+					do_action( 'jetpack_plugin_install_failed', $this->get_plugin_info( $slug ), $errors['code'], $errors['message'] );
 				}
 				return;
 			}
-		}
-
-		if ( ! isset( $details['type'] ) ||
-			'plugin' !== $details['type'] ||
-			is_wp_error( $upgrader->skin->result ) ||
-			! method_exists( $upgrader, 'plugin_info' )
-		) {
-			return;
-		}
-
-		if ( 'install' === $details['action'] ) {
-			$plugin_path = $upgrader->plugin_info();
-			$plugins = get_plugins();
-			$plugin_info = $plugins[ $plugin_path ];
-
 			/**
 			 * Signals to the sync listener that a plugin was installed and a sync action
 			 * reflecting the installation and the plugin info should be sent
 			 *
-			 * @since 4.9.0
+			 * @since 5.8.0
+			 *
+			 * @module sync
 			 *
-			 * @param string $plugin_path Path of plugin installed
-			 * @param mixed $plugin_info Array of info describing plugin installed
+			 * @param array() $plugin, Plugin Data
 			 */
-			do_action( 'jetpack_installed_plugin', $plugin_path, $plugin_info );
+			do_action( 'jetpack_plugin_installed',  array_map( array( $this, 'get_plugin_info' ), $plugins ) );
```

Comment:
Hate to nitpick that late in the process, but do you think we need to keep the old action as a deprecated action?